lib/index.js

```diff
@@ -1,45 +1,25 @@
 const parse = require('./parse');
 const compileFromApiElements = require('./compile');
 
-function createParserErrorCompilationResult(message) {
-  return {
-    mediaType: null,
-    transactions: [],
-    annotations: [{
-      type: 'error', component: 'apiDescriptionParser', message, location: [[0, 1]],
-    }],
-  };
-}
 
-function compile(source, filename, callback) {
-  // All regular parser-related or compilation-related annotations
-  // should be returned in the "compilation result". Callback should get
-  // an error only in case of unexpected crash.
+function compile(apiDescription, filename, callback) {
+  parse(apiDescription, (err, parseResult) => {
+    // Shouldn't happen, 'parse' turns all parser crashes into annotations
```

Comment:
That's not nessecerily correct. Take a look at https://github.com/apiaryio/api-elements.js/issues/37#issuecomment-447159019:

> I believe we settled on a difference between code errors (which throw or reject a promise) vs parsing errors which return a parse result with an error. For example, with the `validate` API, when an ADD is errenous, the validation did succeed and thus it doesn't cause an error. A parse result containing an error is returned.
>
> There is one exception, and that is for C++ logic error in Protagonist from Drafter. This will likely cause a crash of Node and cannot be recovered, segfault and similiar bugs. This shouldn't happen(tm) but it could. We treat these as high severity bugs when discovered. In Drafter.JS (and possibly the future WASM) I believe this type of problem will translate to throwing a JS error.

So tl;dr, we may error when we cannot recover or have some hard-exception thrown in our parsers.

Reply:
But this is my parse, not your parse 😏 

https://github.com/apiaryio/dredd-transactions/pull/229/files#diff-9f97a9c36fbbfb0d19068d6fae3bb84eR45